contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -1467,6 +1467,10 @@ TdsTypeSpatialToDatum(StringInfo buf)
 	 */
 	/* We are copying first 4 Byte SRID from buf */
 	appendBinaryStringInfo(destBuf, buf->data + buf->cursor, 4);
+	/* Swapping first 3 bytes of SRID as the driver expects SRID to be in little endian order
+	 * 4th byte is always 0
```

Comment:
Do we need to set the 4th byte to 0? Are we assuming that the 4th byte of of the SRID in the incoming buf is already 0?

Reply:
PostGIS stores it in this format:
```
uint8_t srid[3]; /* 24 bits of SRID */
```
and moreover SRIDs are in range [0, 999999] which will lie in the range for 3 bytes. so automatically it will use maximum of 3 bytes and 4th byte will also be 0 which will be received from driver. So, no need to set it explicitly